synapse/lib/snap.py

```diff
@@ -1206,6 +1206,18 @@ async def iterNodeEdgesN2(self, buid, verb=None):
                     await edgeset.add(edge)
                     yield edge
 
+    async def hasNodeData(self, buid, name):
+        '''
+        Return True if the buid has nodedata set on it under the given name
+        False otherwise
+        '''
+        for layr in reversed(self.layers):
+            todo = s_common.todo('getNodeData', buid, name)
```

Comment:
Lets add/use a `hasNodeData()` API to the layer that uses `self.dataslab.has()` to check without lifting/returning the value